### PR TITLE
Add initial spaces page

### PR DESCRIPTION
**Pull request by devnook**
This adds a new page to host Twitter Spaces

Preview: https://deploy-preview-8323--web-dev-staging.netlify.app/spaces/

Currently added spaces are just for visual testing of the MVP and should be removed before merging (all dirs under /en/spaces/...)

### CODE REVIEW

File: src/scss/pages/_spaces.scss [hunk 1]
```diff
@@ -0,0 +1,206 @@
+.spaces {
+  --blue-1: #3740ff;
+  --blue-2: #3c82ff;
+  --blue-3: #3fc4ff;
+
+  audio {
+    width: 250px;
```
Comment (jeffposnick) [inline src/scss/pages/_spaces.scss:7]: There are a couple of places in this file where `px` units are used, a few others that use `em`, and then some references to sizes in variables.

That's not necessarily a problem, but I wanted to point it out in case it happened organically and you want to reconcile things.

File: src/site/_filters/calendar-link.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * @return {string} Calendar url string.
+ */
+function calendarLink(date, title, description) {
+  const end = DateTime.fromJSDate(date);
+  const start = end.minus({days: 1});
+  const url = new URL('https://calendar.google.com/calendar/render');
+  url.searchParams.set('action', 'TEMPLATE');
+  url.searchParams.set(
+    'dates',
+    [start.toFormat('yyyyMMdd'), end.toFormat('yyyyMMdd')].join('/'),
```
Comment (jeffposnick) [inline src/site/_filters/calendar-link.js:36]: This is more of a product decision, but is everyone cool with the calendar link creating full-day events, without any start/end times? I can see the most value coming from adding a calendar event that has the actual start/end time during the day, so that I would be reminded to join the space at the right time when it's about to go live.

File: src/site/_includes/spaces.njk [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+{% set spaces = collections['twitter-space'] %}
+<div class="spaces wrapper region all-center">
+  <h2 class="text-size-4  gap-bottom-size-2">{{ 'i18n.spaces.upcoming_spaces' | i18n(locale) }}</h2>
+  <div class="auto-grid all-center">
+
+  {% set upcomingSpaces = spaces | livePosts | filterInUpcoming('event_date') %}
+  {% set pastSpaces = spaces | livePosts | filterOutUpcoming('event_date') %}
+
+  {% if not upcomingSpaces.length %}
+  <p>A friendly message reminding to check for new upcomig spaces soon</p>
```
Comment (jeffposnick) [inline src/site/_includes/spaces.njk:31]: Do you want to put in a **TODO: REPLACE_ME** or something prominent so that you don't accidentally launch with this?

File: src/site/_includes/spaces.njk [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+              {% if host.host %} <span class="pill" data-inactive>{{ 'i18n.spaces.host' | i18n(locale) }}</span> {% endif %}
+            </li>
+          {% endfor %}
+        </ul>
+        <time class="eyebrow">{{ space.data.event_date | prettyDate('DATE_FULL') }}</time>
+        <p class="gap-top-size-1 gap-bottom-size-2">{{ space.data.description }}</p>
+      </div>
+      {% if space.data.twitter_link %}
+        <a class="button" href="{{ space.data.twitter_link }}" target="_blank">
+          {% include "icons/calendar.svg" %}
```
Comment (jeffposnick) [inline src/site/_includes/spaces.njk:61]: Using a "Calendar+" icon with a "See on Twitter" link is a bit confusing.

File: src/site/_includes/spaces.njk [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+          <span>{{ 'i18n.spaces.see_on_twitter' | i18n(locale) }}</span>
+        </a>
+      {% else %}
+        <a class="button" href="{{ space.data.event_date | calendarLink(space.data.title, space.data.description) }}" target="_blank">
```
Comment (jeffposnick) [inline src/site/_includes/spaces.njk:65]: I know that support for alternative calendars can be tough to code, but it's not clear from the link that "Add to Calendar" will only work with Google Calendar.

I'd recommend putting something in the UI to clarify that this functionality will only work with Google Calendar.

File: src/site/_includes/spaces.njk [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    </article>
+    {% endif %}
+  {% endfor %}
+
+  {% for space in upcomingSpaces %}
+    {% if loop.index0 > 5 %}
+    <input type="checkbox" id="see-more-upcoming" hidden>
+    <label id="see-more-upcoming-label" for="see-more-upcoming" class="button">see more</label>
+
+    <article class="card flow hidden">
```
Comment (jeffposnick) [inline src/site/_includes/spaces.njk:79]: From a quick look at the HTML, it seems like this `article` section shares a _lot_ in common with the previous bit.

Are there opportunities to remove that redundancy by refactoring the sub-sections into a reusable component?

It's optional, of course, but it might help with long-term maintenance.

File: src/site/_includes/spaces.njk [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            {{ space.data.title }}
+          </h2>
+          <p>
+            {% for host in space.data.hosts %}
+              <span class="host">{{ host.name }} {% if host.host %} <span class="pill" data-inactive>{{ 'i18n.spaces.host' | i18n(locale) }}</span> {% endif %}</span>
+            {% endfor %}
+          <details class="ellipsis">
+            <summary><span class="ellipsis__full">{{ space.data.description }}</span><span class="ellipsis__more">{{ 'i18n.spaces.see_more' | i18n(locale) }}</span></summary>
+          </details>
+          <p><audio src="{{ space.data.audio }}" class="audio-player__element" controls="" id="audio-player"></audio></p>
```
Comment (jeffposnick) [inline src/site/_includes/spaces.njk:150]: You might end up with multiple `audio` elements on the page with the same `id="audio-player"` attribute.

File: src/site/content/en/patterns/web-vitals-patterns/infinite-scroll/infinite-scroll/assets/body.html [hunk 8]
```diff
@@ -7,6 +7,6 @@
     <div class="item">E</div>
     <button id="infinite-scroll-button" disabled>
         <span class="disabled-text">Loading more items...</span>
-        <span class="active-text">Show more</span>
+        <span class="active-text"></span>
```
Comment (jeffposnick) [inline src/site/content/en/patterns/web-vitals-patterns/infinite-scroll/infinite-scroll/assets/body.html:10]: Was this change intentional? It's not clear that it's related to the rest of the code in this PR.

File: src/site/content/en/spaces/space-x1-test/index.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
+---
+title: Title Goes Here Title Goes Here Title Goes Here
+description: We’re learning all about Vertex AI this week as Carter Morgan and Jay Jenkins Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+tags: twitter-space
+hosts:
+  - name: Name 1 here
+    twitter: devnook
+    image: https://pbs.twimg.com/profile_images/519869363344515072/s8vGg8fE_400x400.jpeg
+    host: true
```
Comment (jeffposnick) [inline src/site/content/en/spaces/space-x1-test/index.md:9]: See previously comment—I think this would be clearer if it was `isPrimary` instead of `host.`

File: src/site/content/en/spaces/space-x1-test/index.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
+---
+title: Title Goes Here Title Goes Here Title Goes Here
+description: We’re learning all about Vertex AI this week as Carter Morgan and Jay Jenkins Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+tags: twitter-space
+hosts:
+  - name: Name 1 here
+    twitter: devnook
+    image: https://pbs.twimg.com/profile_images/519869363344515072/s8vGg8fE_400x400.jpeg
```
Comment (jeffposnick) [inline src/site/content/en/spaces/space-x1-test/index.md:8]: I know that this is just test data, but what's the strategy for displaying Twitter profile images once we have live data?

Hotlinking to hardcoded `pbs.twimg.com` URLs seems like it could be problematic, both because that `400x400.jpeg` URL doesn't seem to be officially supported, and because someone might change their Twitter profile image at some point and that might break existing URLs.

What would the process be for asking folks to provide Twitter profile image URLs when they wanted to create a new event?

It might be safer to rely on the profile images we already have for authors, served from URLs we control, even if their web.dev profile image differs from their Twitter profile image.

Here's Twitter guidance: https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/user-profile-images-and-banners

File: src/site/content/en/spaces/space-x1-test/index.md [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+    host: true
+  - name: Name 2 here
+    twitter: devnook
+    image: https://pbs.twimg.com/profile_images/519869363344515072/s8vGg8fE_400x400.jpeg
+    host: false
+twitter_link: http://twitter.com
+event_date: 2023-02-21
+audio: https://traffic.libsyn.com/secure/thecsspodcast/TCP_CSS_Podcast__Episode_003_v2.0_FINAL.mp3?dest-id=1891556
```
Comment (jeffposnick) [inline src/site/content/en/spaces/space-x1-test/index.md:16]: Similar to my other comment about hotlinking to Twitter profile images, what's the strategy for storing and linking to Twitter Spaces recordings from a URL we control? How will we guide Hosts to obtain the correct URL when they do this for themselves?

**Review by jeffposnick — CHANGES_REQUESTED** [review]

File: src/site/_includes/spaces.njk (hunk 3 above)
Comment (devnook) [reply src/site/_includes/spaces.njk:31]: No need, I expect to update it today once i get the copy.

File: src/site/_includes/spaces.njk [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+      </div>
+      <div class="card__content flow">
+        <h2 class="card__heading text-size-3">
+          {{ space.data.title }}
+        </h2>
+        <ul class="hosts">
+          {% for host in space.data.hosts %}
+            <li>
+                {{ host.name }}
+              {% if host.host %} <span class="pill" data-inactive>{{ 'i18n.spaces.host' | i18n(locale) }}</span> {% endif %}
```
[the remark this answers is not in the document]
Comment (devnook) [reply src/site/_includes/spaces.njk:52]: changed to primary

File: src/site/_includes/spaces.njk (hunk 5 above)
Comment (devnook) [reply src/site/_includes/spaces.njk:65]: I'll ask designers to update the icon/button, but I would prefer not to block on this.

File: src/site/content/en/spaces/space-x1-test/index.md (hunk 10 above)
Comment (devnook) [reply src/site/content/en/spaces/space-x1-test/index.md:8]: Spaces hosts will not necessarily be web.dev authors, this is why we are not using author's images for this avatars. From technical point of view, this can be any img url. I'll work with Spaces creators to get a process established for those pics (e.g. uploading copy to our img pipeline).

File: src/site/content/en/spaces/space-x1-test/index.md (hunk 11 above)
Comment (devnook) [reply src/site/content/en/spaces/space-x1-test/index.md:16]: We will have url we control for these audio files (stored on drive), this one is just for testing purposes.

**Comment by devnook** [comment]
Thanks a lot @jeffposnick , PTAL!

**Review by jeffposnick — APPROVED** [review]
Modulo the few pending comments, this looks 👍 from me.

**Comment by una** [comment]
At around 905px wide, the bottom line of text gets cut off on the left:

Screen Shot 2022-07-14 at 4 15 43 PM

File: src/site/content/en/spaces/space-x1-test/index.md (hunk 10 above)
Comment (una) [reply src/site/content/en/spaces/space-x1-test/index.md]: +1 to using the existing images we have and using web.dev authors yaml here. Can we still update this and go with this approach?
